src/blockdev.rs

```diff
@@ -173,7 +173,22 @@ impl Partition {
     }
 
     pub fn get_holders(&self) -> Result<Vec<String>> {
-        let holders = Path::new("/sys/block")
+        let holders = self.get_sysfs_dir()?.join("holders");
+        let mut ret: Vec<String> = Vec::new();
+        for ent in read_dir(&holders).chain_err(|| format!("reading {}", &holders.display()))? {
+            let ent = ent.chain_err(|| format!("reading {}", &holders.display()))?;
```

Comment:
Maybe something like

```suggestion
            let ent = ent.chain_err(|| format!("iterating {}", &holders.display()))?;
```

? So it's different from the `read_dir` error itself.